runtime/master/src/main/java/edu/snu/nemo/runtime/master/PlanStateManager.java

```diff
@@ -252,18 +308,30 @@ public synchronized void onTaskStateChanged(final String taskId, final TaskState
     final String stageId = RuntimeIdManager.getStageIdFromTaskId(taskId);
     final List<List<TaskState>> taskStatesOfThisStage = stageIdToTaskAttemptStates.get(stageId);
     final long numOfCompletedTaskIndicesInThisStage = taskStatesOfThisStage.stream()
-      .map(attempts -> attempts.stream()
-        .map(state -> state.getStateMachine().getCurrentState())
-        .allMatch(curState -> curState.equals(TaskState.State.COMPLETE)
-          || curState.equals(TaskState.State.SHOULD_RETRY)
-          || curState.equals(TaskState.State.ON_HOLD)))
+      .map(attempts -> {
+        final List<TaskState.State> states = attempts
+          .stream()
+          .map(state -> (TaskState.State) state.getStateMachine().getCurrentState())
+          .collect(Collectors.toList());
+        return states.stream().allMatch(curState -> curState.equals(TaskState.State.ON_HOLD)) // all of them are ON_HOLD
```

Comment:
What if the task to be ON_HOLD is cloned?

Reply:
Then the clone will become READY->EXECUTING->ON_HOLD like the original, and the stage will become COMPLETE.

Reply:
However, the stage will be considered as `Completed` after all cloned tasks become `ON_HOLD`.

Reply:
You're right. I got confused there. I'll change allMatch -> anyMatch like COMPLETE.